Assets/PatchKitTools/Editor/Menus/BuildAndPublish.cs

```diff
@@ -82,44 +81,69 @@ private void OnAppSelected(Api.Models.Main.App app)
             BeginBuildView();
         }
 
-        private void BeginBuildView()
+        private void BeginBuildView() // *** 3 view *** //
         {
-            var build = new Views.BuildApp();
+            Views.BuildApp build = new Views.BuildApp(_selectedApp);
 
             build.OnSuccess += OnBuildSuccess;
             build.OnFailure += OnBuildFailed;
             build.OnChangeApp += OnBuildChangeApp;
 
             _currentView = build;
+
+            messagesView.ClearList();
         }
 
-        private void BeginSelectAppView()
+        private void BeginSelectAppView() // *** 1 view *** //
         {
-            var selectApp = new Views.SelectApp(_api);
+            Views.ChooseApp chooseApp = new Views.ChooseApp();
+
+            chooseApp.OnCreateApp += OnCreateApp;
+            chooseApp.OnSelectApp += OnSelectApp;
+
+            _currentView = chooseApp;
+        }
+
+        private void OnSelectApp() // *** 2.1 view *** //
```

Comment:
The name isn't reflective of what the method is doing.

I can see it's supposed to be called when the `OnSelectApp` event in the `ChooseApp` view is triggered.

How about renaming this method to `BeginSelectAppView`, renaming the existing `BeginSelectAppView` to `BeginSelectOrCreateAppView` and renaming `OnCreateApp` to `BeginCreateAppView`.